dropshot_endpoint/src/api_trait.rs

```diff
@@ -154,13 +157,61 @@ impl ApiMetadata {
     }
 }
 
+/// A mirror of dropshot's `TagConfig`, used as part of arguments to the
+/// top-level `api_description` macro.
+#[derive(Deserialize, Debug)]
+struct ApiTagConfig {
+    #[serde(default)]
+    allow_other_tags: bool,
+    #[serde(default)]
+    endpoint_tag_policy: ApiEndpointTagPolicy,
+    // tag_definitions is required
+    tag_definitions: HashMap<String, ApiTagDetails>,
+}
+
+/// A mirror of dropshot's `EndpointTagPolicy`, used as part of arguments to the
+/// top-level `api_description` macro.
+#[derive(Clone, Copy, Default, Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
```

Comment:
are there other places where we use snake_case in a similar setting?

Reply:
Hmm, I don't think so -- we use snake_case for keys but don't have any other places where enum variants are tagged as values. `snake_case` looks more correct to me here though, do you think otherwise?

Reply:
I guess I don't have an opinion. It seemed odd to me... based on my knowledge of the implementation of serde_tokenstream, not as an end-user. Maybe show it to a couple of unbiased people and see if they have a preference?

Reply:
Ran a poll on Mastodon which overwhelmingly agreed with you: https://hachyderm.io/@rain/112832705113530863

But one suggestion was to use `EndpointTagPolicy::AtLeastOne`, i.e. the actual value in there. I think that should be possible with `ParseWrapper`, right?

Reply:
Implemented this and I really like it, but at what cost: https://github.com/oxidecomputer/serde_tokenstream/pull/195